### PR TITLE
DOC: add example of spinning circle icon animation

### PR DESCRIPTION
**Pull request by joAschauer**
## PR Summary

Adds an example to the animations section showing an animated spinning circle loading icon.

Documentation failed to build locally due to some latex error in the example `misc/multipage_pdf.py`. The added example is working when run individually.

## PR Checklist

**Tests and Styling**
- [N/A ] Has pytest style unit tests (and `pytest` passes).
- [N/A ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [N/A] New plotting related features are documented with examples.

**Release Notes**
- [N/A ] New features are marked with a `.. versionadded::` directive in the docstring and documented in `doc/users/next_whats_new/`
- [N/A ] API changes are marked with a `.. versionchanged::` directive in the docstring and documented in `doc/api/next_api_changes/`
- [N/A ] Release notes conform with instructions in  `next_whats_new/README.rst` or `next_api_changes/README.rst`

### CODE REVIEW

File: examples/animation/spinning_circle.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    colors=np.linspace(0.15, 0.85, 12, dtype=str),
+    capstyle='round',
+)
+
+ax.set_rmax(1.1)  # make round capstyle on outside visible
+
+def update(*args):
+    new_colors = np.roll(lines.get_colors(), shift=-1, axis=0)
+    lines.set_color(new_colors)
+    return lines
```
Comment (tacaswell) [inline examples/animation/spinning_circle.py:31]:
```suggestion
    return lines,
```

The update function should return a sequence of the artists updated so the blitting works.

**Comment by jklymak** [comment]
Can you advocate for this a bit more?   What new does this bring to the already existing examples?  Thanks!

**Comment by tacaswell** [comment]
xref #24274 

Thank you for your work on this @joAschauer !

In general our documentation suffers from both the problem that we do not have enough (there are things that should be documented that are not) and too much (it is hard to actually find the thing you need!).  While this is very cool example, why do you think this helps the first problem (not enough docs) without making the second worse (too much docs)?

I'm a bit worried about the multipdf failures...

**Comment by joAschauer** [comment]
I did not find a combination of Axes.vlines and polar projection in the documentation. Maybe good to show this somewhere, although it is already answered on stackoverflow (https://stackoverflow.com/a/36828129). If you think this does not add enough value and rather clutters the examples, feel free to close the PR.

**Comment by joAschauer** [comment]
Maybe better to make a proper animation tutorial (https://github.com/matplotlib/matplotlib/pull/24274) than adding more and more examples. I leave it up to you if this should be included or not.

**Comment by jklymak** [comment]
@joAschauer Thanks a lot for this.  OTOH, I'm not clear why it is an animation, nor particularly why the static demonstration is needed.  I don't think we need every plotting method repeated in polar co-ordinates, and adding vlines doesn't seem particularly complicated?  But likely I am missing a subtlety?

**Comment by jklymak** [comment]
Moved to draft until we get more explanation.

**Comment by joAschauer** [comment]
Hi @jklymak and sorry for the late reply. 

You are right, adding vlines to a polar plot is not complicated, but I personally like to learn from examples. 

Sometimes I wonder how something would be done in matplotlib and I play around. The same was true for the spinning circles waiting icon on my phone, which is animated. Since I liked the result, I thought it could be shared with others and opened this PR. As I feel you think it goes beyond the scope of the examples and doesn't add enough value, I am going to close this pull request.

**Pull request closed without merging**